### PR TITLE
Clean up various files

### PR DESCRIPTION
**Pull request by sarveshkaushal**
Following clean up:
1. Removed unnecessary semicolon.
2. Wildcard type argument ? explicitly extends Object class
3. Simplified if-else
4. Removed redundant variables

### CODE REVIEW

File: util/src/main/java/io/kubernetes/client/util/Yaml.java [hunk 1]
```diff
@@ -201,7 +201,7 @@ public static void dump(Object object, Writer writer) {
    * @param data The list of YAML API objects
    * @return A String representing the list of YAML API objects.
    */
-  public static String dumpAll(Iterator<? extends Object> data) {
```
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/util/Yaml.java:204]: Now that we have `KubernetesType` I think this should be `? extends KubernetesType`

File: util/src/main/java/io/kubernetes/client/util/Yaml.java [hunk 2]
```diff
@@ -211,7 +211,7 @@ public static String dumpAll(Iterator<? extends Object> data) {
    * @param data The list of YAML API objects.
    * @param output The writer to output the YAML String to.
    */
-  public static void dumpAll(Iterator<? extends Object> data, Writer output) {
```
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/util/Yaml.java:214]: same comment as above.

**Comment by brendandburns** [comment]
one comment.

**Comment by sarveshkaushal** [comment]
I have added the requested change.

File: util/src/main/java/io/kubernetes/client/util/Yaml.java (hunk 1 above)
Comment (sarveshkaushal) [reply util/src/main/java/io/kubernetes/client/util/Yaml.java:204]: Done

File: util/src/main/java/io/kubernetes/client/util/Yaml.java (hunk 2 above)
Comment (sarveshkaushal) [reply util/src/main/java/io/kubernetes/client/util/Yaml.java:214]: Done

**Review by yue9944882 — COMMENTED** [review]
merging as @brendandburns 's comments been addressed

/lgtm
/approve

**Pull request merged**